File: CS/DiagnosticTool/DXTraceEventParser.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

namespace DiagnosticTool.LinuxTools {
    public class DXTraceEventParser {
        object GetPayload(string payload, string name) {
            Regex regex = new Regex(@"(.*)[\\]+""" + Regex.Escape(name) + @"[\\]+"":[\\]*""?(?<payload>[^ \\,""]+)[\\]*""?(.*)");
            return regex.Match(payload).Groups["payload"].Value;
        }
        public DXTraceEvent Parse(string log) {
            string pattern = @"\[(?<Timestamp>[0-9\.:]+)\]" +
            @" \(([0-9\+\.\?]+)\) (\w+) ([\w:]+) { cpu_id = (?<CPU_ID>[0-9]+) }, { vtid = (?<ThreadID>[0-9]+) }," +
            @" { ([\w\d\s=]+), EventName = ""(?<EventName>\w+)"", (\w+) = ""([\w\s-]+)"", Payload = (""""|""{(?<Payload>.+)}"") }";
            Match match = Regex.Match(log, pattern);
            return new DXTraceEvent(
                match.Groups["EventName"].Value,
                DateTime.Parse(match.Groups["Timestamp"].Value),
                name => GetPayload(match.Groups["Payload"].Value, name));
        }

        public void Process(Action<DXTraceEvent> process, string path) {
            using(StreamReader reader = new StreamReader(path)) {
                while(reader.Peek() > 0) {
                    process(Parse(reader.ReadLine()));
                }
            }
        }
    }
}

File: VB/DiagnosticTool/DXTraceEventParser.vb
Imports System
Imports System.IO
Imports System.Text.RegularExpressions

Namespace DiagnosticTool.LinuxTools

    Public Class DXTraceEventParser

        Private Function GetPayload(ByVal payload As String, ByVal name As String) As Object
            Dim regex As Regex = New Regex("(.*)[\\]+""" & Regex.Escape(name) & "[\\]+"":[\\]*""?(?<payload>[^ \\,""]+)[\\]*""?(.*)")
            Return regex.Match(payload).Groups("payload").Value
        End Function

        Public Function Parse(ByVal log As String) As DXTraceEvent
            Dim pattern As String = "\[(?<Timestamp>[0-9\.:]+)\]" & " \(([0-9\+\.\?]+)\) (\w+) ([\w:]+) { cpu_id = (?<CPU_ID>[0-9]+) }, { vtid = (?<ThreadID>[0-9]+) }," & " { ([\w\d\s=]+), EventName = ""(?<EventName>\w+)"", (\w+) = ""([\w\s-]+)"", Payload = (""""|""{(?<Payload>.+)}"") }"
            Dim match As Match = Regex.Match(log, pattern)
            Return New DXTraceEvent(match.Groups("EventName").Value, Date.Parse(match.Groups("Timestamp").Value), Function(name) GetPayload(match.Groups("Payload").Value, name))
        End Function

        Public Sub Process(ByVal pProcess As Action(Of DXTraceEvent), ByVal path As String)
            Using reader As StreamReader = New StreamReader(path)
                While reader.Peek() > 0
                    pProcess(Parse(reader.ReadLine()))
                End While
            End Using
        End Sub
    End Class
End Namespace
